### ISSUE TITLE
Fails on Nextjs project

### ISSUE THREAD

**Issue by jchatrny** [franciscop/check-licenses#3]
```
node node_modules/check-licenses/index.js
Error: These dependencies are missing their package.json:
➤ node_modules/@next/swc-android-arm-eabi/package.json (not found)
➤ node_modules/@next/swc-android-arm64/package.json (not found)
➤ node_modules/@next/swc-darwin-x64/package.json (not found)
➤ node_modules/@next/swc-freebsd-x64/package.json (not found)
➤ node_modules/@next/swc-linux-arm-gnueabihf/package.json (not found)
➤ node_modules/@next/swc-linux-arm64-gnu/package.json (not found)
➤ node_modules/@next/swc-linux-arm64-musl/package.json (not found)
➤ node_modules/@next/swc-linux-x64-gnu/package.json (not found)
➤ node_modules/@next/swc-linux-x64-musl/package.json (not found)
➤ node_modules/@next/swc-win32-arm64-msvc/package.json (not found)
➤ node_modules/@next/swc-win32-ia32-msvc/package.json (not found)
➤ node_modules/@next/swc-win32-x64-msvc/package.json (not found)

To try to solve this you can try:
➤ Perform a normal install with  npm install 
➤ Reinstall all with  rm -rf node_modules && npm install 
```
Tried to reinstall depenencies and error still shows. Option skip those or log them as "unlicensed" would be nice.

**Comment by franciscop** [comment]
Why do those have no package.json? That's very strange, any official package should have a package.json AFAIK, could you do please e.g. `ls node_modules/@next/swc-linux-arm64-musl/`?

**Comment by jchatrny** [comment]
Sure, they do but theres installed package only for my architecture.

```
➜ ls -l  node_modules/@next/swc-darwin-arm64 
total 79952
-rw-r--r--  1 jakubchatrny  staff        88 Oct 23 19:33 README.md
-rw-r--r--  1 jakubchatrny  staff  40805032 Oct 23 19:33 next-swc.darwin-arm64.node
-rw-r--r--  1 jakubchatrny  staff       268 Oct 23 19:33 package.json
➜ cat node_modules/@next/swc-darwin-arm64/package.json 
{
  "name": "@next/swc-darwin-arm64",
  "version": "12.2.5",
  "os": [
    "darwin"
  ],
  "cpu": [
    "arm64"
  ],
  "main": "next-swc.darwin-arm64.node",
  "files": [
    "next-swc.darwin-arm64.node"
  ],
  "license": "MIT",
  "engines": {
    "node": ">= 10"
  }
}
➜  ls -l node_modules/@next
total 0
drwxr-xr-x  7 jakubchatrny  staff  224 Oct 23 19:33 env
drwxr-xr-x  6 jakubchatrny  staff  192 Oct 23 19:33 eslint-plugin-next
drwxr-xr-x  5 jakubchatrny  staff  160 Oct 23 19:33 swc-darwin-arm64
                                                                  
```

It seems that Next has somewhere reference packages of swc for all possible architectures, however only one is installed

```
➜ cat node_modules/next/package.json
{
  "name": "next",
  ...
  "optionalDependencies": {
    "@next/swc-android-arm-eabi": "12.2.5",
    "@next/swc-android-arm64": "12.2.5",
    "@next/swc-darwin-arm64": "12.2.5",
    "@next/swc-darwin-x64": "12.2.5",
    "@next/swc-freebsd-x64": "12.2.5",
    "@next/swc-linux-arm-gnueabihf": "12.2.5",
    "@next/swc-linux-arm64-gnu": "12.2.5",
    "@next/swc-linux-arm64-musl": "12.2.5",
    "@next/swc-linux-x64-gnu": "12.2.5",
    "@next/swc-linux-x64-musl": "12.2.5",
    "@next/swc-win32-arm64-msvc": "12.2.5",
    "@next/swc-win32-ia32-msvc": "12.2.5",
    "@next/swc-win32-x64-msvc": "12.2.5"
  },
  "gitHead": "911ba233d9d12c0a46c87ee62e783b97583fbbd0"
}
```
It uses `optionalDependencies` (I had no idea there's such thing).

---

I have working `--ignore-missing-dependencies` in my branch would you be interested to have PR for such feature?

**Comment by ThomasVanleynseele** [comment]
Having the same issue in my Next.js project, I would be interested in a flag to ignore this.

**Comment by franciscop** [comment]
Do you mind sharing your `package-lock.json`? At least the part relative to e.g. `node_modules/@next/swc-linux-arm64-musl/`? I believe if a package is marked as optional and we don't find the package.json, we can skip it without much fuzz.

About your proposed solution, I actually implemented most of it back in the day apparently, but I see I ended up hardcoding it to require the `package.json`. If you look [at this line](https://github.com/franciscop/check-licenses/blob/master/index.js#L99):

```
  // Find all the required dependencies
  const packages = await findDependencies(process.cwd(), {
    package: "required",
  });
```

That `required` can be set to `skip`, in which case it will ignore the folders without a `package.json`.

**Comment by franciscop** [comment]
Okay no worries, forget my last comment, any style preference? It will not fail with *missing optional dependencies*, but still fail with missing normal dependencies, which solves the issue at hand here while still being strict:

<table>
  <tr><th>Grayed</th><th>Yellow</th></tr>
  <tr>
    <td>
      <img width="500px" src="https://user-images.githubusercontent.com/2801252/199941123-74f4acc5-d0fb-40e0-bb69-1c4b15043603.png" />
    </td>
    <td>
      <img width="500px" src="https://user-images.githubusercontent.com/2801252/199941118-f603a0f6-0e14-4e63-8e4a-25d99b04dee9.png" />
    </td>
  </tr>
  <tr>
    <td>
      <img width="500px" src="https://user-images.githubusercontent.com/2801252/199941105-b23440c5-045a-4cc7-9cc2-80c6a489ef03.png" />
    </td>
    <td>
      <img width="500px" src="https://user-images.githubusercontent.com/2801252/199941054-9f4efdbf-1262-4732-a2f1-ad59e4851e9a.png" />
    </td>
  </tr>
</table>

**Comment by franciscop** [comment]
<img width="400" alt="image" src="https://user-images.githubusercontent.com/2801252/199942227-8517cac9-c5be-423a-a52d-f48fc84c56c5.png">

<img width="404" alt="image" src="https://user-images.githubusercontent.com/2801252/199942348-2c2e2dfe-2671-4a9a-91b3-592e5b69bf96.png">

**Comment by franciscop** [comment]
**Published as `check-licenses@1.1.0`** 🥳🥳

Please feel free to reopen if this is still an issue, or open a new issue if something else comes up!
